src/webgpu/format_info.ts

```diff
@@ -1874,6 +1874,26 @@ export function textureDimensionAndFormatCompatible(
   );
 }
 
+/**
+ * Returns true iff a texture can be created with the provided GPUTextureDimension
+ * (defaulting to 2d) and GPUTextureFormat for a GPU device, by spec.
+ */
+export function textureDimensionAndFormatCompatibleForDevice(
+  device: GPUDevice,
+  dimension: undefined | GPUTextureDimension,
+  format: GPUTextureFormat
+): boolean {
+  if (
+    dimension === '3d' &&
+    ((isBCTextureFormat(format) && !device.features.has('texture-compression-bc-sliced-3d')) ||
+      (isETC2TextureFormat(format) && !device.features.has('texture-compression-etc2-sliced-3d')) ||
```

Comment:
```suggestion
      // This is not a real feature, but if it were, this is what it would be called.
      (isETC2TextureFormat(format) && !device.features.has('texture-compression-etc2-sliced-3d')) ||
```